src/apps/cct.cpp

```diff
@@ -292,8 +292,35 @@ int main(int argc, char **argv) {
     }
 
     /* Setup transformation */
-    P = proj_create_argv (nullptr, o->pargc, o->pargv);
-    if ((nullptr==P) || (0==o->pargc)) {
+    if (o-> pargc == 0 && o->fargc > 0) {
+        /* Assume we got a auth:code combination */
+        std::string input(o->fargv[0]);
+        int n = input.find(":");
+        if (n > 0) {
+            std::string auth = input.substr(0,n);
+            std::string code = input.substr(n+1, input.length());
+            P = proj_create_from_database(
+                    nullptr, auth.c_str(), code.c_str(), PJ_CATEGORY_COORDINATE_OPERATION, 0, nullptr
+            );
+        } else {
```

Comment:
There are a few operation names that include ':' in them. See
```
$ echo "SELECT * FROM coordinate_operation_view WHERE name LIKE '%:%';" | sqlite3 data/proj.db
other_transformation|PROJ|CRS84_TO_EPSG_4326|OGC:CRS84 to WGS 84||EPSG|9843|Axis Order Reversal (2D)|OGC|CRS84|EPSG|4326|0.0|0
other_transformation|PROJ|CRS27_TO_EPSG_4267|OGC:CRS27 to NAD27||EPSG|9843|Axis Order Reversal (2D)|OGC|CRS27|EPSG|4267|0.0|0
other_transformation|PROJ|CRS83_TO_EPSG_4269|OGC:CRS83 to NAD83||EPSG|9843|Axis Order Reversal (2D)|OGC|CRS83|EPSG|4269|0.0|0
concatenated_operation|PROJ|KKJ_TO_ETRS89|KKJ to ETRS89 (using PROJ:YKJ_TO_ETRS35FIN)|Transformation based on a triangulated irregular network||||EPSG|4123|EPSG|4258||0
```
So it would be prudent to do
```suggestion
        }
        if( P == nullptr ) {
```

Reply:
and P should be initialized to nullptr at line 198